Classes/Utility/GeneralUtility.php

```diff
@@ -413,19 +413,21 @@ public static function fullUrl($url, $domain = null)
             if ($url === '/') {
                 $url = '';
             }
-
-            // remove first /
-            if (strpos($url, '/') === 0) {
-                $url = substr($url, 1);
-            }
-
+            
             if ($domain !== null) {
-                // specified domain
-                $url = 'http://' . $domain . '/' . $url;
+                if(!preg_match('/^https?:\/\//i', $domain)) {
+                    $domain = 'http://' . $domain;
```

Comment:
What about https support? Regex expression allows https, but `$domain =` forgets that.

 Btw., is this an old (separate) issue, which always existed (eventually add a separate github issue?)?